### PR TITLE
Update strikethrough replace rule to fix multiple tilde error

### PR DESCRIPTION
**Pull request by akamefi202**
Update strikethrough replace rule to fix multiple tilde error when transition from text to html.
To make sure that `<del>` tag contains content text, if it's wrapped by multiple tilde characters.
Currently, `<del>` tag contains 2 tilde characters too after transition.

### Fixed Issues
https://github.com/Expensify/App/issues/25693

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
In ExpensiMark-HTML-test.js file, test named "Test strikethrough with multiple tilde characters" works correctly. 
2. What tests did you perform that validates your changed worked?
Checked if the updated strikethrough rule only wraps content text inside multiple tilde characters.

# QA
1. What does QA need to do to validate your changes?
Run the new test cases.
3. What areas to they need to test for regressions?
N/A

### CODE REVIEW

File: package-lock.json
[the remark this answers is not in the document]
Comment (akamefi202) [reply package-lock.json:1]: Done

**Comment by akamefi202** [comment]
@parasharrajat I updated the PR title and description.

File: __tests__/ExpensiMark-HTML-test.js [hunk 1]
```diff
@@ -1225,3 +1225,11 @@ test('Test link with code fence inside the alias text part', () => {
         + '[test <pre>code</pre> test](<a href="https://google.com" target="_blank" rel="noreferrer noopener">google.com</a>)';
     expect(parser.replace(testString)).toBe(resultString);
 });
+
+test('Test strikethrough with multiple tilde characters', () => {
+    let testString = '~~~hello~~~';
+    expect(parser.replace(testString)).toBe('~~<del>hello</del>~~');
+
```
Comment (neil-marcellini) [inline __tests__/ExpensiMark-HTML-test.js:1232]: NAB: Remove spaces on blank line

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
I think we should also test html to text and html to markdown to make sure that all works too.

I don't really understand the full regex, so let's be sure to also test the steps from [the PR](https://github.com/Expensify/expensify-common/pull/461) where this was last changed.

File: __tests__/ExpensiMark-HTML-test.js [hunk 2]
```diff
@@ -1225,3 +1225,11 @@ test('Test link with code fence inside the alias text part', () => {
         + '[test <pre>code</pre> test](<a href="https://google.com" target="_blank" rel="noreferrer noopener">google.com</a>)';
     expect(parser.replace(testString)).toBe(resultString);
 });
+
+test('Test strikethrough with multiple tilde characters', () => {
+    let testString = '~~~hello~~~';
+    expect(parser.replace(testString)).toBe('~~<del>hello</del>~~');
+    
+    testString = '~~~~~~';
+    expect(parser.replace(testString)).toBe('~~~~~~');
```
[the remark this answers is not in the document]
Comment (akamefi202) [reply __tests__/ExpensiMark-HTML-test.js]: Okay

**Review by tylerkaraszewski — APPROVED** [review]

**Review by neil-marcellini — APPROVED** [review]

**Comment by akamefi202** [comment]
@tylerkaraszewski @parasharrajat @neil-marcellini Can we merge this PR if it is approved?

**Pull request merged**